spec/src/main/doc/cdi-se.asciidoc

```diff
@@ -0,0 +1,51 @@
+[[cdi-se]]
+
+== Using CDI in Java SE Applications
+
+=== Bootstrapping a container
+
+In Java EE, CDI comes enabled by default in your application, but if you're working with a Java SE application you can bootstrap a CDI container yourself to enable dependency injection.
+
+[source,java]
+----
+public static void main(String... args) {
+    CDIProvider provider = CDI.getCDIProvider();
+    CDI<Object> cdi = container.initialize();
+    // retrieve a bean and do work with it
+    MyBean myBean = cdi.select(MyBean.class).get();
+    myBean.doWork();
+    // when done
+    cdi.shutdown();
+}
+----
+
+The `CDI` utility class provides static lookup for `CDIProvider` instances based on `ServiceLoader` (this doesn't preclude the notion of manually instantiating the provider in a non-portable manner).  By default, the provider will be looked up via a ServiceLoader but can be over-written by calling `setCDIProvider` on the utility.  Once a `CDIProvider` has been located, it should not be instantiated again, instead the same one returned again.
+
+The `CDIProvider` instance returned is capable of starting a `CDI` instance.  By invoking either of the  `initialize` methods it can start up a container to retriever beans, fire events, gain access to the `BeanManager`.  The `initialize` methods on the `CDIProvider` interface must not be invoked when in an EE container.  If they are invoked, an `UnsupportedOperationException` may be thrown.  Likewise, if `shutdown` or `close` were invoked on the `CDI` class while in an EE container, an `UnsupportedOperationException` may also be thrown.
+
+Only a single `CDI` instance may be initialized within an SE application at a time, based on the provider used.  If a container has already been initialized and you call `initialize` again, an `IllegalStateException` is thrown.  If you manually instantiate a `CDIProvider` and initialize it, non-portable behavior may occur.
```

Comment:
I thought we decided to let the possibility to have multiple container in one application. I agree that it is useless without a new bean discovery mechanism but nothing in your proposed API prevent this possibility. Am I wrong?